src/lib/components/quiz/quiz.svelte

```diff
@@ -0,0 +1,418 @@
+  import RightArrow from '$lib/components/icons/right-arrow.svelte';
+  import MoreInfo from './more-info.svelte';
+
+  export let data: QuizData;
+
+  const TEAMS = {
+    [TeamMatch.AI]: acmAI,
+    [TeamMatch.DEV]: acmDev,
+    [TeamMatch.DESIGN]: acmDesign,
+    [TeamMatch.ALGO]: acmAlgo,
+  };
+  const TEAMLESS = { [TeamMatch.TEAMLESS]: acmGeneral };
+  //  [acmAlgo.title, acmDev.title, acmDesign.title, acmAI.title]
+  let index = 0;
+  let responses: TeamMatch[] | undefined;
+  let answeredAllQuestions = false;
+  // local storage stuff
+  let quizStorage: QuizStorage | undefined;
+  $: {
+    answeredAllQuestions =
+      responses?.length === data.questions.length && !responses.includes(undefined);
+    // Updates local storage whenever `responses` changes.
+    quizStorage && quizStorage.setResponses(responses);
+  }
+  let showResults = false;
+  let showMoreInfo = false;
+  let showTeam: AcmPath;
+
+  $: talliedResponses = (responses ?? []).reduce((tallies, match) => {
+    if (tallies[match]) tallies[match]++;
+    else tallies[match] = 1;
+    return tallies;
+  }, {} as Record<TeamMatch, number>);
+
+  $: match = (Object.entries(talliedResponses)
+    .sort(([, a], [, b]) => b - a)
+    ?.shift()
+    ?.shift() ?? TeamMatch.TEAMLESS) as TeamMatch;
+
+  function goLeft() {
+    if (index > 0) index--;
+  }
+
+  function goRight() {
+    if (index < data.questions.length - 1) {
+      index++;
+    }
+  }
+
+  function recordAnswer(match: TeamMatch) {
+    responses[index] = match;
+    goRight();
+  }
+
+  function submitResponses() {
+    // show results and hide the question
+    showResults = true;
+  }
+
+  function restartQuiz() {
+    responses = [];
+    index = 0;
+    showResults = false;
+    quizStorage.clearResponses();
+  }
+
+  function showTeamDetails(currentTeam: AcmPath) {
+    showMoreInfo = true;
+    showTeam = currentTeam;
+  }
+
+  function goBackToResults() {
+    showMoreInfo = false;
+  }
+
+  onMount(() => {
+    // hypothetically change this to QuizStorage.init()
+    quizStorage = new QuizStorage();
+    responses = quizStorage.getResponses();
+    showResults = responses?.length === data.questions.length && !responses.includes(undefined);
+  });
+</script>
+
+<div class="container">
+  <!-- DISPLAY THE QUIZ QUESTIONS -->
+  {#if !showResults}
+    <h1>ACM TEAM QUIZ</h1>
+    <div class="question">
+      <h2>{data.questions[index].prompt}</h2>
+      <section class="answers">
+        {#each data.questions[index].choices as choice (choice.content)}
+          <button
+            on:click={() => recordAnswer(choice.match)}
+            class:selected-response={(responses ?? [])[index] === choice.match}
+          >
+            <h3>{choice.content}</h3>
+          </button>
+        {/each}
+      </section>
+    </div>
+    <div class="arrow-wrapper">
+      <button
+        on:click={goLeft}
+        disabled={index === 0}
+        class={`${index === 0 && 'disable-arrow'} arrow`}><LeftArrow /></button
```

Comment:
I think this button element can contain the class `false`. To fix that, we can use a Svelte shorthand to conditionally assign a CSS class to an element.
```suggestion
        class:disable-arrow={index === 0}
        class="arrow"><LeftArrow /></button
```